pgml-extension/src/orm/dataset.rs

```diff
@@ -380,133 +170,452 @@ pub fn load_breast_cancer(limit: Option<usize>) -> (String, i64) {
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 16].into_datum(),
+                    row.worst_texture.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 17].into_datum(),
+                    row.worst_perimeter.into_datum(),
                 ),
+                (PgBuiltInOids::FLOAT4OID.oid(), row.worst_area.into_datum()),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 18].into_datum(),
+                    row.worst_smoothness.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 19].into_datum(),
+                    row.worst_compactness.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 20].into_datum(),
+                    row.worst_concavity.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 21].into_datum(),
+                    row.worst_concave_points.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 22].into_datum(),
+                    row.worst_symmetry.into_datum(),
                 ),
                 (
                     PgBuiltInOids::FLOAT4OID.oid(),
-                    breast_cancer.data[(i * breast_cancer.num_features) + 23].into_datum(),
+                    row.worst_fractal_dimension.into_datum(),
+                ),
+                (PgBuiltInOids::BOOLOID.oid(), (row.target == 0).into_datum()),
+            ],
+        );
+        inserted += 1;
+    }
+
+    ("pgml.breast_cancer".to_string(), inserted)
+}
+
+#[derive(Deserialize)]
+struct DiabetesRow {
+    age: f32,
+    sex: f32,
+    bmi: f32,
+    bp: f32,
+    s1: f32,
+    s2: f32,
+    s3: f32,
+    s4: f32,
+    s5: f32,
+    s6: f32,
+    target: f32,
+}
+
+pub fn load_diabetes(limit: Option<usize>) -> (String, i64) {
+    Spi::run("DROP TABLE IF EXISTS pgml.diabetes");
+    Spi::run(
+        "CREATE TABLE pgml.diabetes (
+        age FLOAT4, 
+        sex FLOAT4, 
+        bmi FLOAT4, 
+        bp FLOAT4, 
+        s1 FLOAT4, 
+        s2 FLOAT4, 
+        s3 FLOAT4, 
+        s4 FLOAT4, 
+        s5 FLOAT4, 
+        s6 FLOAT4, 
+        target FLOAT4
+    )",
+    );
+
+    let limit = match limit {
+        Some(limit) => limit,
+        None => usize::MAX,
+    };
+
+    let data: &[u8] = std::include_bytes!("datasets/diabetes.csv.gz");
```

Comment:
This is loaded into the `.so`, so something to think about. Our `.so` is already pretty large (~18MB), I am not sure how that impacts performance.